giskard/project.py

```diff
@@ -136,18 +141,22 @@ def _validate_model_type(model_type):
             )
 
     @staticmethod
-    def _validate_input_types(input_types):
-        if input_types and type(input_types) is dict:
-            if not set(input_types.values()).issubset(set(column_type.value for column_type in SupportedColumnType)):
+    def _validate_column_types(column_types, df):
+        if column_types and type(column_types) is dict:
+            if not set(column_types.values()).issubset(set(column_type.value for column_type in SupportedColumnType)):
                 raise ValueError(
-                    f"Invalid input_types parameter: "
+                    f"Invalid column_types parameter: "
                     + f"Please choose types among {[column_type.value for column_type in SupportedColumnType]}."
                 )
         else:
             raise ValueError(
-                f"Invalid input_types parameter: {input_types}. Please specify non-empty dictionary."
+                f"Invalid column_types parameter: {column_types}. Please specify non-empty dictionary."
             )
 
+    @staticmethod
+    def _transform_prediction_function(prediction_function, df, feature_names):
```

Comment:
`df` argument isn't used in this function

Reply:
Why did we need to add this wrapper function? We already apply the same logic of filtering dataset columns with `feature_names` on the backend (check `run_predict` in model.py)
It'll be difficult for users to debug if we change their prediction function on the fly before saving it

Reply:
1) in most ML libraries prediction_function(df) does not work when df has more columns than feature_names (tensorflow, older version of sklearn). In the former version of the library ai-inspector, we wrapped this function for this reason. 
2) Since we use run_predict in the backend that wraps the function, using the same wrapped function in the error handling processes enables to have coherent error handling and avoid cases like GSK-128.
2)Another benefit: in some older version of scikit learn and maybe other ML libraries, we need to provide input feature in the same order as the one in the training process. By asking that feature_names has the right order in the documentation and by wrapping the prediction function, we handle this point.

Reply:
I make `transform_prediction_function` public so that people can try to run exactly the same model function as the one uploaded to Giskard